### DISCUSSION TITLE
탭 아이템을 두 번 탭 했을 때 작동방법에 대해 논의하고 싶습니다.

### DISCUSSION THREAD

**Discussion by JIWON1923** [Team-ShortcutsZip/MacC-Team-HappyAnding#410] [General]
## 요약
- iOS 16에서는 탭 아이템을 두 번 탭했을 때, 루트뷰 + 최 상단으로 이동으로 하는 건 어떨까요?
- 다른 버전에서는 루트뷰로 이동 -> 한번 더 탭 하면 최 상단으로 이동하고 있습니다.

## 현재의 ShortcutsZip Navigation 구조
- iOS 16에서는 Navigation View를 사용하고, 다른 부분에서는 Navigation Stack을 사용하고 있습니다.

## 시도한 방법
- **Scene Phase를 사용하자**
  - 저는 Scene Phase가 active 상태라면 Navigation의 Root View라고 생각했습니다.
  - 그래서 phase == .active일 때 scroll 위치를 조정했고, active가 아닐 때 pop to root 코드를 작성했습니다.
  - 하지만 여기서 문제는 해당 scene phase는 tab bar에 존재하므로 항상 active 상태라는 것입니다..! -> 탭 아이템 터치와 관련된 코드이므로 scene phase 위치를 조정할 순 없었어요)
  - 그런데 항상 active 상태임을 확인할 수 없었어요 (공유드렸던 것 처럼 break point가 작동하지 않고 Print문도 작동하지 않아요)

- **Detail View의 isActive를 활용하자**
  - 각 네비게이션 화면마다 State Binding을 활용하면 되지 않을까 생각했습니다.
  - 하지만..! 저희 depth도 꽤 깊고, 오류가 많이 발생했습니다.. (예기치 않은 뷰의 pop)

## 해결 방법
- 현재 사용하고 있는 Navigation View를 Navigation Stack처럼 커스텀을 하면 문제를 해결할 수 있을 것 같습니다. 
- 하지만 남은 기한 내에 오류 없이 완벽하게 분리하기 어렵습니다.
- 그래서 탭을 두 번 탭 했을 때, 우선은 루트뷰 및 최 상단으로 이동하는 방식으로 진행하는건 어떨지 제안드립니다.

**Comment by HanGyeongjun** [comment]
저는 해당 방법 찬성합니다! 두 번 탭했을때 루트뷰 최상단으로 이동하더라도 ux상의 문제는 크게 없을 거라고 생각합니다
스택에서 다시 분리하기 어려우실텐데 파이팅입니다...
